generator/internal/language/templates/rust/crate/src/enum.mustache

```diff
@@ -14,3 +14,16 @@ See the License for the specific language governing permissions and
 limitations under the License.
 }}
 {{> ../../common/enum}}
+
+{{! TODO: before merging, make the following code conditional on the need for the enum to be used as a path parameter }}
+
+impl gax::path_parameter::PathParameter for {{Name}} {
+
+    type P = {{Name}};
+
+    fn required<'a>(&'a self, _: &str) -> std::result::Result<&'a Self::P, gax::path_parameter::Error> {
+        Ok(self)
+    }
+
+
+}
```

Comment:
You can save yourself some trouble, I think this would make enums work as both query and path parameters:

```suggestion
impl gax::request_parameter::RequestParameter for {{Name}} {
    fn format(&self) -> gax::request_parameter::Result {
        Ok(self.0.clone())
    }
}
```

Reply:
Neat! This will require making the `gax::request_parameter` and `gax::request_parameter::RequestParameter` as public though.

I'll look into that test and update the PR later today.